docs/spec/overview.md

```diff
@@ -1,70 +1,114 @@
-# Resource Groups
+# Overview
 
-Knative Eventing API is grouped into _Channels_, _Feeds_ and _Flows_:
+In the serverless model, computation and storage can be intertwined by having
+storage and compute systems publish events which are handled asynchronously by
+further compute services. Event delivery provides common infrastructure for
+decoupling event producers and consumers such that the linking between them can
+be performed without change to either. Loosely coupled eventing systems enable
+composable services which retain the ability to scale independently.
+Declarative event binding reduces the effort to create a scalable system that
+combines managed services and custom code.
 
-* _Channels_ define an abstraction between the eventing infrastructure and the
-  consumption and production of the events.
+Knative eventing implements common components of an event delivery ecosystem:
+enumeration and discovery of event sources, configuration and management of
+event transport, and declarative binding of events (generated either by storage
+services or earlier computation) to further event processing and persistence.
 
-* _Feeds_ bridge the event source into the eventing framework.
+The following is a detailed specification for the `eventing.knative.dev` API
+surface, which provides control mechanism for Knative eventing.
 
-* _Flows_ abstract the path of an event from a source takes to reach an event
-  consumer.
-
-Typically, _Feeds_ perform out-of-cluster provisioning, while _Channels_ are
-within-cluster management of event delivery. _Flows_ is a higher order wrapper
-used to connect event producers directly with event consumers leveraging 
-_Feeds_ and _Channels_.
 
 # Resource Types
 
-The primary resources in the Knative Eventing _Feeds_ API are EventSource,
-EventType and Feed:
+The eventing API defines several resource types as well as interfaces which may
+be fulfilled by other Kubernetes objects ([Sinkable](interfaces.md#sinkable),
+[Subscribable](interfaces.md#subscribable),
+[Channelable](interfaces.md#channelable),
+[Targetable](interfaces.md#targetable)). The API defines and provides a
+complete implementation for [Subscription](spec.md#kind-subscription), and
+abstract resource definitions for [Sources](spec.md#kind-source),
+[Channels](spec.md#kind-channel), and [Providers](spec.md#kind-provisioner)
+which may be fulfilled by multiple backing implementations (much like the
+Kubernetes Ingress resource).
+
+ * A **Subscription** describes the transformation of an event (via a
+   _Callable_) and optional storage of a returned event. 
+
+ * A **Source** represents an ongoing selection of events from an external
+   system. A Subscription is used to connect these events to subsequent
+   processing steps.
+
+ * A **Channel** provides event storage and fanout of events from a well-known
+   input address to multiple outputs described by Subscriptions. 
```

Comment:
maybe bit of a nit here. but the way we describe these resources here are:
 - Describes, represents and provides
it seems a bit of mismatch. Channel says what it does, not what it represents. I think it would be better to try to rephrase all these as logical constructs in the system instead of what they do.
 Source: represents incoming events from an external system (and how to subscribe to them [subscribable])
 Subscription: represents an (optional) event transformation from a Source via Callable for (optionally) further processing via Channels [subscribable].
 Channel: Eventing middleware that can be subscribed to by 0 or more Subscriptions for processing.

Also, I'd like to see little clearer that Source / Channel are very symmetrical in the system from the control plane perspective. Source is Subscribable just like Channel. So Source shoves things into a Sinkable just like Subscription (possibly if so configured) shoves things into SInkable. The way I read this section, Source is somehow special from consumption part and it's really not.

Reply:
I gave this a shot but it might be something we have to do a follow up on with more detail when we work this model out a bit more.